shared/react-native/ios/KeybaseShare/ShareViewController.m

```diff
@@ -329,6 +336,74 @@ - (void)processItem:(NSItemProvider*)item lastItem:(BOOL)lastItem {
+    [[[PushNotifier alloc] init] localNotification:@"extension" msg:@"Sharing text not yet implemented."
+                                        badgeCount:-1 soundName:@"default" convID:@"" typ:@"chat.extension"];
+    [self maybeCompleteRequest:lastItem];
+  };
+  
+  // If we get an image in the form of a UIImage, we first write it to a temp file
+  // and run it through the normal file sharing code path.
+  NSItemProviderCompletionHandler imageHandler = ^(UIImage* image, NSError* error) {
+    NSString* guid = [[NSProcessInfo processInfo] globallyUniqueString];
+    NSString* filename = [NSString stringWithFormat:@"%@%@.jpg", NSTemporaryDirectory(), guid];
+    NSURL* url = [NSURL fileURLWithPath:filename];
+    
+    NSData* imageData = UIImageJPEGRepresentation(image, 0.7);
+    [imageData writeToFile:filename atomically:YES];
+    [self handleFileURL:url item:item lastItem:lastItem];
+    
+    [[NSFileManager defaultManager] removeItemAtPath:filename error:&error];
+    if (error != nil) {
+      NSLog(@"unable to remove temp file: %@", error);
+    }
+  };
+  
+  // The NSItemProviderCompletionHandler interface is a little tricky. The caller of our handler
+  // will inspect the arguments that we have given, and will attempt to give us the attachment
+  // in this form. For files, we always want a file URL, and so that is what we pass in.
+  NSItemProviderCompletionHandler fileHandler = ^(NSURL* url, NSError* error) {
+    // Check for no URL (it might have not been possible for the OS to give us one)
+    if (url == nil) {
+      if ([item hasItemConformingToTypeIdentifier:@"public.image"]) {
+        // Try to handle with our imageHandler function
+        [item loadItemForTypeIdentifier:@"public.image" options:nil completionHandler:imageHandler];
+      } else {
+        [self maybeCompleteRequest:lastItem];
+      }
+      return;
+    }
+    [self handleFileURL:url item:item lastItem:lastItem];
+  };
+  
+  if ([item hasItemConformingToTypeIdentifier:@"public.movie"]) {
+    [item loadItemForTypeIdentifier:@"public.movie" options:nil completionHandler:fileHandler];
+  } else if ([item hasItemConformingToTypeIdentifier:@"public.image"]) {
+    [item loadItemForTypeIdentifier:@"public.image" options:nil completionHandler:fileHandler];
+  } else if ([item hasItemConformingToTypeIdentifier:@"public.file-url"]) {
+    [item loadItemForTypeIdentifier:@"public.file-url" options:nil completionHandler:fileHandler];
+  } else if ([item hasItemConformingToTypeIdentifier:@"public.text"]) {
+    [item loadItemForTypeIdentifier:@"public.text" options:nil completionHandler:textHandler];
+  } else if ([item hasItemConformingToTypeIdentifier:@"public.url"]) {
+    [item loadItemForTypeIdentifier:@"public.url" options:nil completionHandler:urlHandler];
+  } else {
+    [[[PushNotifier alloc] init] localNotification:@"extension" msg:@"We failed to share your content. Please try from the Keybase app."
```

Comment:
Does this happen for types that we can't handle? It seems it'd be better just say we can't share it, as the "upload" feature in app supports less types I think.

Reply:
This happens for incoming file sources that we don't understand, irrespective of file type.

Reply:
I'll change the string.